apis/core.oam.dev/v1alpha1/garbagecollect_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/oam-dev/kubevela/pkg/oam"
+)
+
+const (
+	// GarbageCollectPolicyType refers to the type of garbage-collect
+	GarbageCollectPolicyType = "garbage-collect"
+)
+
+// GarbageCollectPolicySpec defines the spec of configuration drift
+type GarbageCollectPolicySpec struct {
+	// KeepLegacyResource if is set, outdated versioned resourcetracker will not be recycled automatically
+	// outdated resources will be kept until resourcetracker be deleted manually
+	KeepLegacyResource bool `json:"keepLegacyResource,omitempty"`
+
+	// Rules defines list of rules to control gc strategy at resource level
+	// if one resource is controlled by multiple rules, first rule will be used
+	Rules []GarbageCollectPolicyRule `json:"rules"`
+}
+
+// GarbageCollectPolicyRule defines a single garbage-collect policy rule
+type GarbageCollectPolicyRule struct {
+	Selector GarbageCollectPolicyRuleSelector `json:"selector"`
+	Strategy GarbageCollectStrategy           `json:"strategy"`
+}
+
+// GarbageCollectPolicyRuleSelector select the targets of the rule
+type GarbageCollectPolicyRuleSelector struct {
+	TraitTypes []string `json:"traitTypes"`
+}
```

Comment:
can they be omit-empty?

Reply:
rules can be empty, selector and strategy cannot, fixed it.